### ISSUE TITLE
Loading error no longer leaves file open

### ISSUE THREAD

**Issue by miloush** [sumatrapdfreader/sumatrapdf#3595]
Up to v3.1.2:

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/10546952/092ce13c-9b54-47c1-8e56-985acf5c55fe)

Since v3.2:

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/10546952/1e9fe9fc-5fa1-4aa8-9658-1afaff7ad583)

The difference is that previously, the file was open (and had its own tab if tabs are enabled) and therefore several commands were available, most importantly "Open in Adobe Reader".  So when SumatraPDF is your default PDF reader, you open a PDF file that is only compatible with Adobe Reader (say from temporary internet files), you can proceed with re-opening it in Adobe Reader. This is now incredibly much more difficult.

Furthermore, when opening several files at once, you could see which ones were invalid, while now when tabs are enabled, the error messages just fly through and disappear and you cannot read anything.

I suggest the previous behavior to be restored and files that have loading errors stay open/having their tab.

("Show in Folder" should also work for such files)

**Comment by GitHubRulesOK** [comment]
I have to do some serious damage to a PDF to not see a poor one 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/426f6272-c9b7-4e74-b265-af9c31940043)

However you should be able to send that to Acrobat if it is a PDF whichever interface is active
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/e95671d4-a6de-425a-8339-29671e093ea0)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/efbfb62a-c984-41cd-89a5-1fef5faf246e)

**Comment by miloush** [comment]
You could just try to open non-PDF file to test this :)

Hm, none of those options are available to me:

<img src="https://github.com/sumatrapdfreader/sumatrapdf/assets/10546952/57936d91-4115-4e10-8cd0-23e6ed01bbb8">

But also I can see in your screenshot that the files do get opened for you (you get tabs and in the second case you have the error inside the window, apart from the notification). Maybe your loading fails later than for my files.

**Comment by GitHubRulesOK** [comment]
There are several things to consider so 
- My Acrobat is installed in registry so SumatraPDF will Automatically find it but that may not always be the case
- My Version is Current 3.5 Pre-Release since that is the only version that can be changed, there can be no changes to 3.4.6
- The cause of a file type fail loading may change for different file contents or locations

**Comment by miloush** [comment]
The v3.1.2 is happy with letting me open the same file on the same machine in Adobe Reader:

<img src="https://github.com/sumatrapdfreader/sumatrapdf/assets/10546952/442c1043-9190-4e41-8d1a-f762b0b35a65">

The files I have to open are encrypted and protected by the FileOpen plugin. You can try on the test file they provide, [installcomplete.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/12214259/installcomplete.pdf).

(If you prefer getting it directly from the source, there is a link under the first FAQ entry at https://www.fileopen.com/products/troubleshooting-faq). I can reproduce the reported difference in behavior using this file.

**Comment by miloush** [comment]
I think your test file is good enough to make it to `WndProcCanvasLoadError`, which is not my case. My case seems to behave the same way as when you try to open a docx/zip file for example.

**Comment by GitHubRulesOK** [comment]
Thats not a PDF but an Adobe Custom installer so loads oter applications (SumatraPDF will not do that)
Here we can see the "PDF" is not recognisable as such (Much like JS enabled AI files or XFA which is XML Wolf in a PDF sheepskin) If I open in Acrobat it highjacks my Browser to open a foreign site in MS Edge (that should not be allowed as its a security exploit). Anyway SumatraPDF will clearly open those Tabs that are allowed even the Bad one then finaly show last failed attempt to be exploited.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/024a737e-478f-4ced-8514-e7a739cebba5)

**Comment by miloush** [comment]
Yes, and my point is that files which are "PDFs not recognisable as such" opened in tabs/windows of SumatraPDF before and allowed them to be reopened in Adobe Reader, while since v3.2 that is no longer the case.

**Comment by GitHubRulesOK** [comment]
Ok I see your point since there is no loaded file object (per the error) there is nothing valid to pass on in that case.
However that is not what "Open In" was designed to do if it was as you suggest  a by-products of legacy behaviour it was more by accident than design.

**Comment by miloush** [comment]
As far as I can tell it is actually a valid PDF using Security Handler specified in the Filter entry as per the PDF specification. Adobe Acrobat checks online what to do with that Filter value and gets sent to the webpage that it opens, so not really a security exploit.

I didn't want to make this too complicated, but it sounds like it would be reasonable to detect that the files uses unknown Filter and offer opening it in Acrobat if installed. I just thought it would be easier to not care and always open the file with an error in the canvas.

That way you also avoid users not seeing the error notification when it is replaced with another one or disappears too quickly.

**Comment by miloush** [comment]
I am guessing this commit https://github.com/sumatrapdfreader/sumatrapdf/commit/b247508908c07bbb5bdad0c4f5c4b5f5929e885d broke it by returning early, skipping the call to what used to be `LoadDocIntoCurrentTab`

**Comment by GitHubRulesOK** [comment]
@kjk 
is this regression an easy detection like xfa ?
the error condition seems to block using the file option to pass to an external handler
and secondary issue is that a sidebar issue that favorites is not initiated ?
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/33c27b3d-e57d-4d44-bbef-b1f2c0f09a40)
when the error message is dismissed **it should but does not** look like this
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/b4795fde-0ac5-412f-9c57-87c6d90f5b9c)

**Comment by kjk** [comment]
yeah, that's not trivial to fix

**Comment by kpym** [comment]
It would be helpful to have the ability to send an error report for "Error loading ..." issues. 

In my case, I received a Google invoice (a supposedly valid PDF) that can't be opened with Sumatra but works fine in Chrome and Edge. However, I couldn't find a way to send an error report for you to investigate the issue. _(I cannot share the PDF here as it contains sensitive information.)_

**Comment by GitHubRulesOK** [comment]
@kpym 
I see many different problem PDFs each day some are accepted by all readers some by a few some by none.
So there is no easy way to tell why a fault is allowed in one application and not another unless by tracing the behaviour often in reverse from the faulty behaviour "observed" many will be faulty without observed misbehaviour.

For SumatraPDF to handle a generic fault such as font or truncation or many other "fault groups" a sample is needed for tracing where the "fail" is triggered. And thus the only "private" option is you email the developer kkowalczyk@gmail.com 

For  one common means to try a simplified "fix" in some cases open a PDF in Acrobat Reader MAY trigger its editor into adjusting the file and then on closure it will as if you want to save the changes (fixes). This is the simplest way I know apart from a dedicated re write of rogue contents which is NOT save as in a browser as that usually just preserves any faults.

A common means to regenerate a PDF is to simply pass through GhostScript and check there were no known losses.
roughly  `GSWIN##C -sDEVICE=pdfwrite -oOut.pdf -f In .pdf`

**Comment by kpym** [comment]
Thanks @GitHubRulesOK for your answer.

> For SumatraPDF to handle a generic fault. Such as font or truncation or many other "fault groups" a sample is needed for tracing where the "fail" is triggered. And thus the only "private" option is you email the developer [kkowalczyk@gmail.com](mailto:kkowalczyk@gmail.com)

I'll send my PDF to this email, but I think that for this project it will be nice if in case of this kind of errors we can generate and send a report directly from Sumatra.
 
> For one common means to try a simplified "fix" in some cases open a PDF in Acrobat Reader MAY trigger its editor into adjusting the file and then on closure it will ask if you want to save the changes (fixes). This is the simplest way I know apart from a dedicated re write of rogue contents. Which is NOT "save as" in a browser since that usually just preserves any faults.
> 
> A common means to regenerate a PDF is to simply pass through GhostScript (which will attempt to restructure fonts) and check there were no known losses. roughly `GSWIN##C -sDEVICE=pdfwrite -oOut.pdf -f In .pdf` or more compatible `MuTool Clean -gg in.pdf -o out.pdf`

I know how to clean a PDF, thanks. If I write here is to try to make Sumatra better, not just to solve my problem ;)

**Comment by GitHubRulesOK** [comment]
@kjk I can still open NON PDF Standard files in any other PDF viewer such as 3.1.2 or Acrobat or Exchange
They don't have to be installed but can be loaded by OR added to external viewers, same as before, (only custom Icons are blocked)

<img width="289" height="70" alt="Image" src="https://github.com/user-attachments/assets/caf42eb9-d265-4e58-a978-b3f248a90315" />

Considering the issue is that such files be "handed off" to other more suited viewer is the OP issue, this is not a problem with 3.7 pre-release 

<img width="1387" height="714" alt="Image" src="https://github.com/user-attachments/assets/c20d67b7-ddc4-41fb-8d72-5b22124689ad" />
